### ISSUE TITLE
[Feat]: Support for older browsers (ios 12)

### ISSUE THREAD

**Issue by wiegell** [coder/code-server#4785]
ES2020 is a showstopper for some older browsers e.g. all IOS 12 devices (original iPad air in my case). Safari chokes on optional chaining (?.)

ES6 compile was tested in https://github.com/coder/vscode/pull/23, but then removed again in [#d4f09b4](https://github.com/coder/vscode/commit/d4f09b4df0d23ead4389b4a69c6fad86ac358892).
Any plans to reinstate ES6?

For other people on ios 12, the latest code-server still on ES6 (before microsoft bumped vscode to ES2020) is code-server@3.10.2

**Comment by jsjoeio** [comment]
I don't think so unfortunately. We did test it but then had to remove it because it broke Jupyter Notebooks. I don't know if there's a solution that doesn't cause something to break. If you ideas though, we'd be open to it

**Comment by wiegell** [comment]
What about an argument for the install script that gives the option of building with ES6?

**Comment by jsjoeio** [comment]
I like the sound of that! But my concern is then we'd be shipping a version of code-server that only partially works since Jupyter Notebooks would be broken. We'd also have to maintain an additional set of binaries, right?

A couple options:
- I wonder if we could add an option to the install script called "build from source" which could take a config where you could specify "bundle with ES6
- you could fork code-server, make the changes in your fork and then create a build yourself (easiest, preferred solution)

**Comment by wiegell** [comment]
Option one is the neatest while option two might be the easiest. I think it would be ok with some stuff not working as long as the user is sufficiently warned.
My personal incentive have waned, since i now bought a newer tablet. Maybe the issue should just linger to see if its worth the effort?

**Comment by jsjoeio** [comment]
> My personal incentive have waned, since i now bought a newer tablet. Maybe the issue should just linger to see if its worth the effort?

Haha definitely! I'll add it to the Backlog Candidates.

**Comment by helojo** [comment]
is the same problem? my iPad os version is iOS 13.
when I open the url on the ipad browser, there is no any display, and the screen is always white after input the password.But it's ok on Macos or windows.Has anyone else encountered this problem?

![IMG_E9AAAA813D13-1](https://user-images.githubusercontent.com/5379090/161175964-55d80820-9a8a-4be0-a792-5443328e3936.jpeg)

**Comment by wiegell** [comment]
> is the same problem? my iPad os version is iOS 13.
> 
> when I open the url on the ipad browser, there is no any display, and the screen is always white after input the password.But it's ok on Macos or windows.Has anyone else encountered this problem?
> 
> 
> 
> ![IMG_E9AAAA813D13-1](https://user-images.githubusercontent.com/5379090/161175964-55d80820-9a8a-4be0-a792-5443328e3936.jpeg)
> 
> 

As mentioned above - revert to 3.10.2

**Comment by helojo** [comment]
> As mentioned above - revert to 3.10.2

@wiegell thank you,I will upgrade iOS to 15.4.1
